### PR TITLE
Fix for issue #297

### PR DESCRIPTION
**Pull request by bennetty**
#297 
partnering pr on astilectron will be linked

### CODE REVIEW

File: window.go [hunk 1]
```diff
@@ -531,3 +532,14 @@ func (w *Window) Unmaximize() (err error) {
 	_, err = synchronousEvent(w.ctx, w, w.w, Event{Name: EventNameWindowCmdUnmaximize, TargetID: w.id}, EventNameWindowEventUnmaximize)
 	return
 }
+
+// UpdateCustomOptions unmaximize the window
```
Comment (asticode) [inline window.go:536]: Can you update the comment?

**Review by asticode — CHANGES_REQUESTED** [review]
PR is almost perfect.

Just a minor change though

File: window.go [hunk 2]
```diff
@@ -531,3 +532,14 @@ func (w *Window) Unmaximize() (err error) {
...
+func (w *Window) UpdateCustomOptions(o WindowCustomOptions) (err error) {
+	if err = w.ctx.Err(); err != nil {
+		return
+	}
+	w.m.Lock()
+	w.o.Custom = &o
+	w.m.Unlock()
+	return w.w.write(Event{WindowOptions: w.o, Name: EventNameWindowCmdUpdateCustomOptions, TargetID: w.id})
```
Comment (asticode) [inline window.go:544]: Why not use `synchronuousEvent`?

Comment (bennetty) [reply window.go:544]: when testing with synchronousEvent, the event becomes blocking and prevent subsequent calls from being called

Comment (asticode) [reply window.go:544]: Mmm makes sense. But `synchronousEvent` should still be used. To fix the problem:

- JS should write an event `window.event.updated.custom.options` [after this line](https://github.com/asticode/astilectron/blob/master/index.js#L280) (check out [here](https://github.com/asticode/astilectron/blob/master/index.js#L119) to see how to use `write`)
- GO should use `synchronousEvent` and use the `windowEventUpdatedCustomOptions` event you created:

```go
synchronousEvent(..., ..., ..., Event{Name: EventNameWindowCmdUpdateCustomOptions, TargetID: w.id}, EventNameWindowEventUpdatedCustomOptions)
```


I've already merged the JS PR but you can open a new one.

Comment (bennetty) [reply window.go]: Understood! will open a new JS PR

File: window.go [hunk 3]
```diff
@@ -54,6 +55,7 @@ const (
 	EventNameWindowEventDidGetRedirectRequest         = "window.event.did.get.redirect.request"
 	EventNameWindowEventWebContentsExecutedJavaScript = "window.event.web.contents.executed.javascript"
 	EventNameWindowEventWillNavigate                  = "window.event.will.navigate"
+	EventNameWindowEventUpdateCustomOptions           = "window.event.update.custom.options"
```
Comment (asticode) [inline window.go:58]: Could you replace `update` with `updated` in the event name and the variable name?

**Review by asticode — CHANGES_REQUESTED** [review]

**Comment by asticode** [comment]
FYI I've created a `v0.22.0` tag

**Pull request merged**